### PR TITLE
Add a bundle 'jdk-reflection' 

### PR DESCRIPTION
**Pull request by centic9**
This allows to forbid certain reflection calls which are usually blocked by SecurityManagers, more might be useful, but this is the bare minimum set that often indicates invalid access.

### CODE REVIEW

**Comment by uschindler** [comment]
Cool thanks! I will look into merging it. We also collected some bad stuff which might be added as bundles.

**Comment by uschindler** [comment]
I general, I would remove the Method#invoke stuff. The bad stuff is only setAccessible(). Reflection is not bad per se, but working around security is a problem.

File: src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-reflection.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+java.lang.reflect.AccessibleObject#setAccessible(java.lang.reflect.AccessibleObject[], boolean) @ Reflection usage fails with SecurityManagers and likely will not work any more in Java 9
+java.lang.reflect.AccessibleObject#setAccessible(boolean) @ Reflection usage fails with SecurityManagers and likely will not work any more in Java 9
+java.lang.reflect.Method#invoke(java.lang.Object, java.lang.Object[]) @ Reflection usage fails with SecurityManagers and likely will not work any more in Java 9
```
Comment (uschindler) [inline src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-reflection.txt]: As said on issue, I would remove this. It is not bad and won't fail with security managers, unless very restructive. But setAccessible() is a no-go.

**Comment by centic9** [comment]
Sure, I have updated the pull request accordingly now.

**Comment by uschindler** [comment]
Thanks. Currently I am thinking about the fact, if the reflection bundle should be made dependent on Java version like unsafe or deprecated.

One small thing. The test class should be named Java6* instead of Oracle*. The reason is that the antunit/Oracle* classes are compiled only on JVMs that are Oracle. The test is basically a test class that should compile with any JVM. I can change that before regenerating the class file.

**Comment by uschindler** [comment]
I merged this. Will do some cleanups and adding docs in a minute.

**Pull request merged**